.github/workflows/teamcity-services-diff-check-weekly.yml

```diff
@@ -37,11 +37,13 @@ jobs:
           run: echo "GOOGLE_BETA_REPO_PATH=${{ env.OUTPUT_PATH}}" >> $GITHUB_ENV
         - name: Check that new services have been added to the TeamCity configuration code
           run: |
-              # Create lists of service packages in providers
-              ls ${{env.GOOGLE_REPO_PATH}}/google/services > tools/teamcity-diff-check/services_ga.txt
-              ls ${{env.GOOGLE_BETA_REPO_PATH}}/google-beta/services > tools/teamcity-diff-check/services_beta.txt
-    
+              # Create lists of service packages in providers. Need to cd into repos where go.mod is to do this command.
+              cd ${{env.GOOGLE_REPO_PATH}}
+              go list -f '{{.Name}}' ${{env.GOOGLE_REPO_PATH}}/google/services/... > $GITHUB_WORKSPACE/provider_services_ga.txt
```

Comment:
Using go list allows us to list the actual packages present in the services directory. This protects us a bit against people who name the folders differently from the package name (matching names is what is usual).